Code/GraphMol/FileParsers/MolWriters.h

```diff
@@ -150,21 +153,24 @@ class SDWriter : public MolWriter {
   //! \brief flush the ostream
   void flush() {
     PRECONDITION(dp_ostream, "no output stream");
-    dp_ostream->flush();
-  };
-
-  //! \brief close our stream (the writer cannot be used again)
-  void close() {
-    PRECONDITION(dp_ostream, "no output stream");
     try {
       dp_ostream->flush();
     } catch (...) {
+      try {
+        dp_ostream->setstate(std::ios::badbit);
+      } catch (const std::runtime_error& e) {
+      }
     }
+  };
+
+  //! \brief close our stream (the writer cannot be used again)
+  void close() {
+    flush();
     std::ostream *tmp_ostream = dp_ostream;
     dp_ostream = NULL;
     if (df_owner) {
-      delete tmp_ostream;
       df_owner = false;
+      delete tmp_ostream;
```

Comment:
Excuse my ignorance. I'd forgotten about deleting NULL were safe in C++. I was also checking for possibility of double deletes, so I meant to say "undefined pointer" but it looks like the destructor is the only other place that calls delete dp_ostream (which this code sets to NULL).

so, close() followed by a ~destructor() looks safe, and we should pretty much never even need to call destructors explicitly unless we start doing insane placement new stuff (c++11)

this was more for my own sanity check.